### PR TITLE
Remove dependency to psycopg2 with dump/restore

### LINKED ISSUE #35906: postgresql_db dump requires psycopg2
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

postgresql_db

##### ANSIBLE VERSION

```
ansible 2.4.2.0
  config file = /home/user/Documents/ansible_backups/ansible.cfg
  configured module search path = [u'/home/user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.13 (default, Jan 03 2017, 17:41:54) [GCC]
```

##### CONFIGURATION

```
ANSIBLE_PIPELINING(/home/user/Documents/ansible_backups/ansible.cfg) = True
ANSIBLE_SSH_ARGS(/home/user/Documents/ansible_backups/ansible.cfg) = -q -o Compression=yes -o ControlMaster=auto -o ControlPersist=5m -o CheckHostIP=no -o UserKnownHostsFile=/dev/null
ANSIBLE_SSH_CONTROL_PATH(/home/user/Documents/ansible_backups/ansible.cfg) = %(directory)s/ansible-ssh-%%h-%%r
COLOR_VERBOSE(/home/user/Documents/ansible_backups/ansible.cfg) = bright blue
DEFAULT_FORKS(/home/user/Documents/ansible_backups/ansible.cfg) = 50
DEFAULT_HOST_LIST(/home/user/Documents/ansible_backups/ansible.cfg) = [u'/home/user/Documents/ansible_backups/inventory.py']
HOST_KEY_CHECKING(/home/user/Documents/ansible_backups/ansible.cfg) = False
```

##### OS / ENVIRONMENT

openSuse Tumbleweed

##### SUMMARY

When using `state=dump` with postgresql_db module, it fails with the following error:
```
fatal: [my_host]: FAILED! => {"changed": false, "msg": "the python psycopg2 module is required"}
```
However, looking at the source, nothing indicates that psycopg2 is required, since only `psql` is invoked for dumping.

`HAS_PSYCOPG2` should only be tested if the `state` is not `dump` or `restore`

### PR DESCRIPTION
**Pull request by Glandos**
##### SUMMARY

'dump' and 'restore' state only need pg_dump and pg_restore. These tools 
don't use psycopg2 so this change tries to avoid the use of it in these 
cases.

This obsoletes #35907 that fixes #35906.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

postgresql_db

##### ADDITIONAL INFORMATION

The db_exists test was replaced with an error detection when piping to 
compression program, using a FIFO file. This effectively reverts #39483, 
that was a fix for #39412.

### CODE REVIEW

**Comment by Andersson007** [comment]
@Glandos thank you for fixing. I saw it and remember.
When I find free time I'll necessary review it or ask someone to do it.

**Comment by Andersson007** [comment]
And, please, add a changelog fragment there "ansible/changelogs/fragments/", https://docs.ansible.com/ansible/devel/community/development_process.html#making-your-pr-merge-worthy

**Review by Andersson007 — CHANGES_REQUESTED** [review]
I tried it out, works correctly.
Besides a changelog fragment,

1. a typo here:
```
284        # Use a fifo to be notfied of an error in pg_dump
```
2. and add info to the doc (at least, in the Notes block) about that psycopg2 is not necessarily to be installed for dump/restore only.. or something like this.

File: lib/ansible/modules/database/postgresql/postgresql_db.py
Comment (Andersson007) [inline lib/ansible/modules/database/postgresql/postgresql_db.py]:
```suggestion
- State C(dump) and C(restore) don't require I(psycopg2) since version 2.8.
```

**Review by Andersson007 — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/database/postgresql/postgresql_db.py
Comment (Andersson007) [reply lib/ansible/modules/database/postgresql/postgresql_db.py]: Seems that's all, waiting for your feedback.

**Comment by Andersson007** [comment]
hanging windows tests looks vary strange ;) maybe close and reopen the PR to run the tests again ?

**Review by Andersson007 — APPROVED** [review]
shipit

**Comment by Andersson007** [comment]
Needs one more review and shipit by another contributor.
Could anybody look at this PR? @antoinell ?

**Comment by Andersson007** [comment]
@Glandos , would be cool, if you look at #https://github.com/ansible/ansible/pull/52077 and make a couple of important notes ;) or, please, look at another PR what you prefer.
It'll help to improve postgresql-specific support to the next 2.8 release.

We have Postgresql working group here https://github.com/ansible/community/wiki/PostgreSQL
If you want to to be in the working group, just write yourself in a table or let me know I add you.

And we also have a community pinboard here https://github.com/ansible/ansible/pull/52077.
Welcome!

**Comment by Glandos** [comment]
The community pinboard is at ansible/community#435. I'm already subscribed to it.

**Comment by Andersson007** [comment]
@andytom , could you please look at this fix

**Comment by andytom** [comment]
Had a quick look and LGTM.
shipit

**Comment by Andersson007** [comment]
@mattclay could you merge it manually because it has changelog fragment

**Comment by gundalow** [comment]
@Glandos Thank you for this bug fix
@Andersson007 @andytom Thank you for the reviews.

As long as PRs get the `shipit` label added https://github.com/ansible/ansible/pull/53323#event-2193275718 then they will get merged within the next few days.

Merged into `devel` for release in Ansible 2.8

**Pull request merged**
